gpytorch/kernels/rbf_kernel.py

```diff
@@ -90,9 +90,8 @@ def __init__(
         )
 
     def forward(self, x1, x2, **params):
-        x1_ = x1.div(self.lengthscale)
-        x2_ = x2.div(self.lengthscale)
-        x1_, x2_ = self._create_input_grid(x1_, x2_, **params)
-
-        diff = (x1_ - x2_).norm(2, dim=-1)
-        return diff.pow(2).div_(-2).exp_()
+        center = x1.mean(dim=-2, keepdim=True)
+        x1_ = (x1 - center).div(self.lengthscale)
+        x2_ = (x2 - center).div(self.lengthscale)
```

Comment:
Sorry - last change: could we move the centering code into `_covar_sq_distance`?

Reply:
Good point, fixed